internal/clientconn/conninfo/conn_info_test.go

```diff
@@ -72,3 +75,87 @@ func TestConnInfo(t *testing.T) {
 		})
 	}
 }
+
+func TestConnInfoCursorParallelWork(t *testing.T) {
+	t.Parallel()
+
+	connInfo := NewConnInfo()
+
+	runs := runtime.GOMAXPROCS(-1) * 10
+	wg := sync.WaitGroup{}
+	start := make(chan struct{})
+	ready := make(chan struct{}, runs)
+
+	// Test parallel set of cursor.
+	for i := 0; i < runs; i++ {
+		wg.Add(1)
+
+		go func(i int) {
+			defer wg.Done()
+
+			ready <- struct{}{}
+
+			<-start
+			connInfo.SetCursor(fmt.Sprintf("cursor %d", i), nil)
```

Comment:
```suggestion
			connInfo.SetCursor(fmt.Sprintf("cursor %d", i), nil)
			connInfo.Cursor(fmt.Sprintf("cursor %d", i))
```

Just because why not :) - to have more operations concurrently